src/com/amazon/ion/impl/bin/IonManagedBinaryWriter.java

```diff
@@ -1103,7 +1105,7 @@ public void writeBytes(byte[] data, int off, int len) throws IOException
 
     public void flush() throws IOException
     {
-        if (getDepth() == 0 && !user.hasAnnotations())
+        if (getDepth() == 0 && (localsLocked || lstAppendEnabled))
```

Comment:
I think we should keep the `!user.hasAnnotations()` check in here.

Reply:
re-added